nrf_security/src/mbedcrypto_glue/vanilla/vanilla_glue.cmake

```diff
@@ -29,6 +29,10 @@ nrf_security_debug("######### Creating vanilla noglue library #########")
 #
 add_library(${IMAGE}mbedcrypto_vanilla)
 
+if (CONFIG_SHORT_WCHAR_T)
+  target_compile_options(${IMAGE}mbedcrypto_vanilla PRIVATE -fshort-wchar)
+endif()
```

Comment:
Is there anything wrong with doing

target_link_libraries(mbedcrypto_vanilla PRIVATE zephyr_interface)

?

If there is a problem with this then we can at least use "zephyr_get_compile_options_for_lang".

Then we can get rid of this patch and the TOOLCHAIN_C_FLAGS patch below (after also applying the zephyr_compile_options patch mentioned above).